### PR TITLE
spi config to secret

### PR DESCRIPTION
**Pull request by sparkoo**
spi config moved to secret. you have to manually create it because it contains secret data so we can't commit it.

It also creates Route with generated host.

Issue: https://issues.redhat.com/browse/SVPI-48

to test:
1. deploy everything with `./hack/bootstrap-cluster.sh`
2. you should see spi deployed but pods failing to start due to missing mount
3. update `components/spi/config.yaml` with your values (you need to create GitHub oauth app first)
4. create actual secret in cluster with `kubectl create secret generic oauth-config --from-file=components/spi/config.yaml -n spi-system`
5. in few moments, you should see pods running
6. follow https://github.com/redhat-appstudio/service-provider-integration-operator#go-through-the-oauth-flow-manually to test oauth flow

❗  Links to my fork branch are there for testing purpose. I'll revert them back to infra-deployment repo before the merge

### CODE REVIEW

**Comment by skabashnyuk** [comment]
Tested on RHPDS.
Works for me 
Знімок екрана 2022-01-24 о 15 34 33

File: argo-cd-apps/app-of-apps/all-applications-staging.yaml [hunk 1]
```diff
@@ -11,7 +11,8 @@ spec:
 
   source:
     path: argo-cd-apps/overlays/staging
-    repoURL: https://github.com/redhat-appstudio/infra-deployments.git
+    repoURL: https://github.com/sparkoo/infra-deployments.git
```
Comment (mmorhun) [inline argo-cd-apps/app-of-apps/all-applications-staging.yaml:14]: Looks like you forgot to [revert](https://github.com/redhat-appstudio/infra-deployments#development-mode-for-your-own-clusters) development changes

**Comment by Michkov** [comment]
Update README.md with instruction how to set.

File: argo-cd-apps/app-of-apps/all-applications-staging.yaml (hunk 1 above)
Comment (sparkoo) [reply argo-cd-apps/app-of-apps/all-applications-staging.yaml:14]: I'm keeping it here so reviewers can test it. I'll remove them before the merge. Or should I do it now?

**Review by skabashnyuk — APPROVED** [review]

File: argo-cd-apps/overlays/development/repo-overlay.yaml [hunk 2]
```diff
@@ -40,5 +40,5 @@ metadata:
   name: spi
 spec:
   source:
-    repoURL: https://github.com/redhat-appstudio/infra-deployments.git
-    targetRevision: main
+    repoURL: https://github.com/sparkoo/infra-deployments.git
+    targetRevision: svpi48-config
```
Comment (Michkov) [inline argo-cd-apps/overlays/development/repo-overlay.yaml:44]: no such branch

File: components/spi/kustomization.yaml [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
 images:
   - name:  quay.io/redhat-appstudio/service-provider-integration-operator
     newName: quay.io/redhat-appstudio/service-provider-integration-operator
-    newTag: v0.2.0
+    newTag: v0.2.1
   - name: quay.io/redhat-appstudio/service-provider-integration-oauth
     newName: quay.io/redhat-appstudio/service-provider-integration-oauth
-    newTag: v0.2.0
+    newTag: v0.2.1
```
Comment (Michkov) [inline components/spi/kustomization.yaml:14]: https://quay.io/repository/redhat-appstudio/service-provider-integration-oauth?tab=tags the image is tagged `0.2.1` without `v`

File: components/spi/kustomization.yaml [hunk 4]
```diff
@@ -1,16 +1,16 @@
 resources:
-  - https://github.com/redhat-appstudio/service-provider-integration-operator/config/default?ref=ccc1d8a0b400aa3e47c10fa14040674252d3ae3d
-
+  - https://github.com/sparkoo/service-provider-integration-operator/config/default?ref=c36682a4fd7dbac2e7303637e0cc4c5ff958cfd4
```
Comment (Michkov) [inline components/spi/kustomization.yaml:2]: use redhat-appstudio org, the commit is there

File: argo-cd-apps/app-of-apps/all-applications-staging.yaml (hunk 1 above)
Comment (sparkoo) [reply argo-cd-apps/app-of-apps/all-applications-staging.yaml]: fixed

File: argo-cd-apps/overlays/development/repo-overlay.yaml (hunk 2 above)
Comment (sparkoo) [reply argo-cd-apps/overlays/development/repo-overlay.yaml]: fixed

File: components/spi/kustomization.yaml (hunk 3 above)
Comment (sparkoo) [reply components/spi/kustomization.yaml]: fixed

File: components/spi/kustomization.yaml (hunk 4 above)
Comment (sparkoo) [reply components/spi/kustomization.yaml]: fixed

**Comment by sparkoo** [comment]
> Update README.md with instruction how to set.

fixed

File: components/spi/config.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+sharedSecret: <shared-secret>
```
Comment (Michkov) [inline components/spi/config.yaml:1]: what is sharedSecret? That could be just random string, right?

Comment (sparkoo) [reply components/spi/config.yaml:1]: yes, any random string

**Review by sbose78 — APPROVED** [review]

**Pull request merged**